src/lib/connections/bridge

```diff
@@ -25,6 +25,16 @@ bridge_up() {
     done
 
     bring_interface_up "$Interface"
+
+    if [[ ! -z "$MACAddressOf" ]]; then
+      if is_interface "$MACAddressOf"; then
+        ip link set dev "$Interface" address "$(cat /sys/class/net/$MACAddressOf/address)"
```

Comment:
`$(cat` has an equivalent, preferable internal: `$(<`.

So this is executed after the interface is brought up and before it is configured with an IP-address. Is this the correct place, or should it be done before the interface is brought up?

Part of me thinks that we should really only do this whole thing if we are adding an interface (I don't like using an existing interface all that much anyway, but with a bridge it may make sense). In that case, the MAC-address can be included in the argument to `interface_add` along the lines of `$(MACAddressOf:+address $(< /sys/class/net/$MACAddressOf/address))`.

As it is now, I think it can also be achieved by adding `IPCustom="link set dev $interface address $(< ...)"` to the profile. Granted, that is a lot less nice than having the dedicated variable.

Reply:
I changed it around a bit.
Firstly, moving the entire conditional block to the top since we probably don't want anything to be modified before `return 1` is executed.
Then, I incorporated your recommendation of the `$(< ...)` change and the use of `IPCustom` together to simplify the logic a bit.